src/test/team_scoreboard.test.ts

```diff
@@ -0,0 +1,296 @@
+import assert from "assert";
+import GuildPreference from "../structures/guild_preference";
+import Team from "../structures/team";
+import TeamScoreboard from "../structures/team_scoreboard";
+import Player from "../structures/player";
+
+const FIRST_TEAM_NAME = "kmq team";
+const SECOND_TEAM_NAME = "not kmqer";
+
+const FIRST_USERID = "12345";
+const TWO_USER_IDS = [FIRST_USERID, "23456"];
+const FOUR_USER_IDS = [FIRST_USERID, "23456", "252525", "1000000"];
+
+const USER_TAG = "unused";
+const AVATAR_URL = "avatarurl";
+
+let scoreboard: TeamScoreboard;
+
+/* eslint-disable prefer-arrow-callback */
+beforeEach(function () {
+    scoreboard = new TeamScoreboard();
+    scoreboard.addTeam(FIRST_TEAM_NAME, new Player("user#0101", FIRST_USERID, AVATAR_URL, 0));
+});
+
+describe("add a team", function () {
+    it("should add the team to the scoreboard", function () {
+        const player = new Player("second_user#1010", TWO_USER_IDS[1], AVATAR_URL, 0);
+        const team = new Team(SECOND_TEAM_NAME, player);
+
+        assert.strictEqual(scoreboard.hasTeam(team.getName()), false);
+        scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        assert.deepStrictEqual(scoreboard.getTeam(team.getName()), team);
+        assert.strictEqual(scoreboard.hasTeam(team.getName()), true);
+        assert.deepStrictEqual(Object.values(scoreboard.getTeams()), [new Team(FIRST_TEAM_NAME, new Player("user#0101", FIRST_USERID, AVATAR_URL, 0)), team]);
+    });
+});
+
+describe("get team of player", function () {
+    it("should get the team that corresponds to the player", function () {
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(FOUR_USER_IDS[0]), scoreboard.getTeam(FIRST_TEAM_NAME));
+        assert.strictEqual(scoreboard.getTeamOfPlayer(FOUR_USER_IDS[1]), null);
+
+        const player = new Player("second_user#1010", FOUR_USER_IDS[1], AVATAR_URL, 0);
+        scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(player.getId()), scoreboard.getTeam(SECOND_TEAM_NAME));
+
+        const anotherPlayer = new Player(USER_TAG, FOUR_USER_IDS[2], AVATAR_URL, 0);
+        scoreboard.addPlayer(SECOND_TEAM_NAME, anotherPlayer);
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(anotherPlayer.getId()), scoreboard.getTeam(SECOND_TEAM_NAME));
+
+        const bestPlayer = new Player(USER_TAG, FOUR_USER_IDS[3], AVATAR_URL, 0);
+        scoreboard.addPlayer(FIRST_TEAM_NAME, bestPlayer);
+        scoreboard.removePlayer(bestPlayer.getId());
+        scoreboard.removePlayer(player.getId());
+        scoreboard.removePlayer(FIRST_USERID);
+        assert.strictEqual(scoreboard.getTeam(FIRST_TEAM_NAME), null);
+        assert.deepStrictEqual(Object.values(scoreboard.getTeams()), [new Team(SECOND_TEAM_NAME, anotherPlayer)]);
```

Comment:
same here

Reply:
this entire block doesn't seem to correspond with the test `should get the team that corresponds to the player`